xapian-core/cluster/cluster.cc

```diff
@@ -110,18 +121,16 @@ TermListGroup::add_document(const Document &document) {
 }
 
 void
-TermListGroup::add_documents(DocumentSource &docs) {
+TermListGroup::add_documents(MSetDocumentSource docs) {
```

Comment:
The key words here are "later on". And yes, you certainly want to pass the superclass if you want to support that. However for now I'd just pass an `MSet` and be done with it. We can add something that takes a suitable `Document` iterator in future.